src/relay/backend/aot_executor_codegen.cc

```diff
@@ -1052,7 +1075,22 @@ class AOTExecutorCodegen : public MixedModeVisitor {
     // Retrieve the return sids
     return_sid_ = final_aot_allocator.GetReturnIds();
     // Insert outputs to main func signature
-    CreateIOVar(lowered_main_func->body, "output");
+    // If output tensor names were provided use them
+    if (auto opt = func->GetAttr<Array<String>>("output_tensor_names")) {
+      Array<String> output_tensor_names = opt.value();
+      if (lowered_main_func->body->IsInstance<TupleNode>()) {
+        Tuple output_tuple = Downcast<Tuple>(lowered_main_func->body);
+        for (unsigned i = 0; i < output_tuple->fields.size(); i++) {
+          CreateIOVar(output_tuple->fields[i], output_tensor_names[i]);
```

Comment:
we might want to also enforce use_unique_name here--since if we ever implemented GetOutputIndex on AOTExecutor, the user would need to confirm that the output var didn't get its name mangled. wdyt?

Reply:
Good catch! changed now.